### ISSUE TITLE
error using `@[flag]` in `vlib/flag/flag.v` 

### ISSUE THREAD

**Issue by larpon** [vlang/v#21843]
### Describe the bug

I'm trying to contribute https://github.com/larpon/flags to vlib. 
I'm blocked by a weird error where I can not use the `@[flag]` attribute in the `flag` module.

### Reproduction Steps

Add this code anywhere in `vlib/flag/flag.v`:

```v
@[flag]
pub enum MyEnum {
	one
	two
	three
}
```
then try and run an app or test that imports the `flag` module:
`v test vlib/flag/`

### Expected Behavior

Compile/run succesfully

### Current Behavior

```
  v test vlib/flag/
---- Testing... -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 FAIL  [1/3] C:   217.1 ms, R:   128.174 ms vlib/flag/usage_example_test.v
         retry: 1
      comp_cmd: '/home/lmp/v/v' -skip-running   -o '/dev/shm/v_1000/tsession_77384b002740_01J2GE0RD0Z1T546Y2QD76BRYJ/2_2/usage_example_test' '/home/lmp/v/vlib/flag/usage_example_test.v'
       run_cmd: '/dev/shm/v_1000/tsession_77384b002740_01J2GE0RD0Z1T546Y2QD76BRYJ/2_2/usage_example_test'
failure code: 1; foutput.len: 524; failure output:
---------------------------------------------------------------------------------------------------- retry: 0 ; max_retry: 0 ; r.exit_code: 1 ; trimmed_output.len: 356
/home/lmp/v/vlib/flag/usage_example_test.v:11: ✗ fn testsuite_begin
   > assert res.exit_code == 0
     Left value: 1

/home/lmp/v/vlib/flag/usage_example_test.v:26: ✗ fn check_program
   > assert res.exit_code == 0
     Left value: 127
 FAIL  [2/3] C:   217.4 ms, R:   137.321 ms vlib/flag/default_flag_options_test.v
         retry: 1
      comp_cmd: '/home/lmp/v/v' -skip-running   -o '/dev/shm/v_1000/tsession_77384b002740_01J2GE0RD0Z1T546Y2QD76BRYJ/0_0/default_flag_options_test' '/home/lmp/v/vlib/flag/default_flag_options_test.v'
       run_cmd: '/dev/shm/v_1000/tsession_77384b002740_01J2GE0RD0Z1T546Y2QD76BRYJ/0_0/default_flag_options_test'
failure code: 1; foutput.len: 538; failure output:
---------------------------------------------------------------------------------------------------- retry: 0 ; max_retry: 0 ; r.exit_code: 1 ; trimmed_output.len: 370
/home/lmp/v/vlib/flag/default_flag_options_test.v:11: ✗ fn testsuite_begin
   > assert res.exit_code == 0
     Left value: 1

/home/lmp/v/vlib/flag/default_flag_options_test.v:24: ✗ fn check_program
   > assert res.exit_code == 0
     Left value: 127
 FAIL  [3/3] C:   839.3 ms, R:     0.000 ms vlib/flag/flag_test.v
>> compilation failed:
vlib/flag/flag.v:5:38: error: duplicate of a module name `flag`
    3 | // data object storing information about a defined flag
    4 | pub struct Flag {
    5 | pub:
      |     ^
    6 |     name     string // name as it appears on command line
    7 |     abbr     u8     // shortcut
vlib/flag/flag.v:6:38: error: duplicate of a module name `flag`
    4 | pub struct Flag {
    5 | pub:
    6 |     name     string // name as it appears on command line
      |                                         ~~~~
    7 |     abbr     u8     // shortcut
    8 |     usage    string // help message
vlib/flag/flag.v:7:38: error: duplicate of a module name `flag`
    5 | pub:
    6 |     name     string // name as it appears on command line
    7 |     abbr     u8     // shortcut
      |                                ^
    8 |     usage    string // help message
    9 |     val_desc string // something like '<arg>' that appears in usage,
vlib/flag/flag.v:9:40: error: duplicate of a module name `flag`
    7 |     abbr     u8     // shortcut
    8 |     usage    string // help message
    9 |     val_desc string // something like '<arg>' that appears in usage,
      |                                           ~~~~
   10 |     // and also the default value, when the flag is not given
   11 | }
vlib/flag/flag.v:11:41: error: duplicate of a module name `flag`
    9 |     val_desc string // something like '<arg>' that appears in usage,
   10 |     // and also the default value, when the flag is not given
   11 | }
      |  ^
   12 | 
   13 | struct UnknownFlagError {

-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Failed command 1:    '/home/lmp/v/v' -skip-running   -o '/dev/shm/v_1000/tsession_77384b002740_01J2GE0RD0Z1T546Y2QD76BRYJ/2_2/usage_example_test' '/home/lmp/v/vlib/flag/usage_example_test.v'
Failed command 2:    '/home/lmp/v/v' -skip-running   -o '/dev/shm/v_1000/tsession_77384b002740_01J2GE0RD0Z1T546Y2QD76BRYJ/0_0/default_flag_options_test' '/home/lmp/v/vlib/flag/default_flag_options_test.v'
Failed command 3:    '/home/lmp/v/v' -skip-running   -o '/dev/shm/v_1000/tsession_77384b002740_01J2GE0RD0Z1T546Y2QD76BRYJ/1_1/flag_test' '/home/lmp/v/vlib/flag/flag_test.v'
Summary for all V _test.v files: 3 failed, 3 total. Elapsed time: 1014 ms, on 3 parallel jobs. Comptime: 1273 ms. Runtime: 265 ms.
```

### V version

V 0.4.6 304aa9f

### Environment details (OS name and version, etc.)

EndeavourOS (rolling, Arch-based) linux
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.
